File: src/Entity/Ordonnance.php
<?php

namespace App\Entity;

use App\Repository\OrdonnanceRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
#[ORM\Entity(repositoryClass: OrdonnanceRepository::class)]
class Ordonnance
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(type: "date")]
    #[Assert\NotNull(message: "La date est obligatoire.")]
    #[Assert\Type("\DateTimeInterface")]
  
    private ?\DateTimeInterface $date = null;
    

    #[ORM\Column(length: 255)]
    #[Assert\Length(max: 255, maxMessage: "Les médicaments ne peuvent pas dépasser 255 caractères.")]
    #[Assert\NotBlank(message: "Le champ des médicaments ne peut pas être vide.")]
    private ?string $medicaments = null;

    #[ORM\Column(length: 255)]
    #[Assert\Length(max: 500, maxMessage: "Le commentaire ne peut pas dépasser 500 caractères.")]
    private ?string $commantaire = null;

    #[ORM\Column(length: 255)]
    #[Assert\Length(max: 255, maxMessage: "La durée d'utilisation ne peut pas dépasser 255 caractères.")]
    #[Assert\NotBlank(message: "La durée d'utilisation est obligatoire.")]
    private ?string $duree_utilisation = null;

    #[ORM\Column(length: 255)]
    #[Assert\Length(max: 255, maxMessage: "La quantité d'utilisation ne peut pas dépasser 255 caractères.")]
    #[Assert\NotBlank(message: "La quantité d'utilisation est obligatoire.")]
    private ?string $quantite_utilisation = null;

    #[ORM\OneToOne(cascade: ['persist', 'remove'],orphanRemoval: true)]
    #[ORM\JoinColumn(name: 'id_consultation_id', referencedColumnName: 'id', onDelete: 'CASCADE')]
    private ?Consultation $id_consultation = null;

    public function getId(): ?int
    {
        return $this->id;
    }
    public function __toString(): string
    {
        return (string) $this->id; // Ou toute autre propriété que vous souhaitez afficher
    }
    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): static
    {
        $this->date = $date;

        return $this;
    }

    public function getMedicaments(): ?string
    {
        return $this->medicaments;
    }

    public function setMedicaments(string $medicaments): static
    {
        $this->medicaments = $medicaments;

        return $this;
    }

    public function getCommantaire(): ?string
    {
        return $this->commantaire;
    }

    public function setCommantaire(string $commantaire): static
    {
        $this->commantaire = $commantaire;

        return $this;
    }

    public function getDureeUtilisation(): ?string
    {
        return $this->duree_utilisation;
    }

    public function setDureeUtilisation(string $duree_utilisation): static
    {
        $this->duree_utilisation = $duree_utilisation;

        return $this;
    }

    public function getQuantiteUtilisation(): ?string
    {
        return $this->quantite_utilisation;
    }

    public function setQuantiteUtilisation(string $quantite_utilisation): static
    {
        $this->quantite_utilisation = $quantite_utilisation;

        return $this;
    }

    public function getIdConsultation(): ?Consultation
    {
        return $this->id_consultation;
    }

    public function setIdConsultation(?Consultation $id_consultation): static
    {
        $this->id_consultation = $id_consultation;

        return $this;
    }
}

File: src/main/java/org/example/entities/Ordonnance.java
package org.example.entities;

import java.time.LocalDate;

public class Ordonnance {

    private int id;
    private LocalDate date;
    private String medicaments;
    private String commantaire;
    private String dureeUtilisation;
    private String quantiteUtilisation;
    private String texteQuantite; // Nouveau champ pour le texte de la quantité

    // Constructeur sans arguments
    public Ordonnance() {}

    // Constructeur avec tous les attributs
    public Ordonnance(Integer id, LocalDate date, String medicaments, String commantaire,
                      String dureeUtilisation, String quantiteUtilisation, String texteQuantite) {
        this.id = id;
        this.date = date;
        this.medicaments = medicaments;
        this.commantaire = commantaire;
        this.dureeUtilisation = dureeUtilisation;
        this.quantiteUtilisation = quantiteUtilisation;
        this.texteQuantite = texteQuantite; // Initialisation du texte de la quantité
    }

    // Constructeur avec id
    public Ordonnance(Integer id) {
        this.id = id;
    }

    // Getters et Setters pour chaque attribut

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public String getMedicaments() {
        return medicaments;
    }

    public void setMedicaments(String medicaments) {
        this.medicaments = medicaments;
    }

    public String getCommantaire() {
        return commantaire;
    }

    public void setCommantaire(String commantaire) {
        this.commantaire = commantaire;
    }

    public String getDureeUtilisation() {
        return dureeUtilisation;
    }

    public void setDureeUtilisation(String dureeUtilisation) {
        this.dureeUtilisation = dureeUtilisation;
    }

    public String getQuantiteUtilisation() {
        return quantiteUtilisation;
    }

    public void setQuantiteUtilisation(String quantiteUtilisation) {
        this.quantiteUtilisation = quantiteUtilisation;
    }

    public String getTexteQuantite() {
        return texteQuantite;
    }

    public void setTexteQuantite(String texteQuantite) {
        this.texteQuantite = texteQuantite; // Setter pour texteQuantite
    }

    @Override
    public String toString() {
        return "Ordonnance{" +
                "id=" + id +
                ", date=" + date +
                ", medicaments='" + medicaments + '\'' +
                ", commantaire='" + commantaire + '\'' +
                ", dureeUtilisation='" + dureeUtilisation + '\'' +
                ", quantiteUtilisation='" + quantiteUtilisation + '\'' +
                ", texteQuantite='" + texteQuantite + '\'' + // Ajout du texteQuantite dans toString
                '}';
    }
}
